### PR TITLE
added support for loading ecmaScriptModules

### PR DESCRIPTION
**Pull request by bergos**

### CODE REVIEW

File: ecmaScriptModule.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  const code = await import(filename)
+
+  if (!method) {
+    return code
+  }
+
+  // split method name by . for deep structures
+  return method.split('.').reduce((code, property) => code[property], code)
+}
```
Comment (tpluscode) [inline ecmaScriptModule.js:22]: This and the `ecmaScript.js` look awfully similar. I would consider combining them

File: package.json [hunk 2]
```diff
@@ -20,15 +20,16 @@
...
   },
   "devDependencies": {
-    "@rdfjs/data-model": "^1.1.2",
+    "@rdfjs/data-model": "^1.2.0",
     "@rdfjs/dataset": "^1.0.1",
-    "@rdfjs/parser-n3": "^1.1.2",
-    "jest": "^24.9.0",
+    "@rdfjs/parser-n3": "^1.1.4",
+    "c8": "^7.3.1",
+    "codecov": "^3.7.2",
```
Comment (tpluscode) [inline package.json:30]: Slightly unrelated, I typically don't install this as dependency but rathe call `npx codecov` in CI

File: test/arguments.test.js [hunk 3]
```diff
@@ -1,135 +1,120 @@
...
-      const result = await loader({ term, dataset }, { loaderRegistry })
+      const result = await loader({ term, dataset }, { loaderRegistry: dummyLoaderRegistry })
 
-      // then
-      expect(result).toEqual(['a', '5'])
+      deepStrictEqual(result, ['a', '5'])
     })
 
-    test('should handle boolean false, but not undefined values from loaders', async () => {
-      // given
```
Comment (tpluscode) [inline test/arguments.test.js:36]: 😢

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: ecmaScriptModule.js (hunk 1 above)
Comment (bergos) [reply ecmaScriptModule.js:22]: I also looked twice, but didn't came up with an easy solution to make one sync the other async. I would like to keep the `require` based implementation backwards compatible.

File: test/arguments.test.js (hunk 3 above)
Comment (bergos) [reply test/arguments.test.js:36]: If it's about the comments: Sorry, but for me, the tests are way more readable without them, when they are reduced to only a few lines. I'm full with you to keep the code of the test itself well structured and be picky if they are not. We can break up bigger tests with support functions or, if that's too tricky, we can add the comments only for those tests, but please let's keep only the code pattern for the average tests.

**Review by tpluscode — APPROVED** [review]

**Pull request merged**